### PR TITLE
docs: update experimental-features doc with IPNS over pubsub changes.

### PR DESCRIPTION
**Pull request by aschmahmann**
Updates the IPNS over PubSub experimental feature to encompass the latest changes to the feature.

### CODE REVIEW

File: docs/experimental-features.md [hunk 1]
```diff
@@ -439,20 +439,22 @@ When it is enabled:
...
 - IPNS resolvers subscribe to the name-specific topic on first
-  resolution and receive subsequently published records through pubsub in real time. This makes subsequent resolutions instant, as they are resolved through the local cache. Note that the initial resolution still goes through the DHT, as there is no message history in pubsub.
+  resolution and receive subsequently published records through pubsub in real time.
+  This makes subsequent resolutions instant, as they are resolved through the local cache.
 
 Both the publisher and the resolver nodes need to have the feature enabled for it to work effectively.
 
+Note: While IPNS pubsub has been available since 0.4.14, it received major changes in 0.5.0.
+Users interested in this feature should upgrade to at least 0.5.0
+
```
Comment (aschmahmann) [inline docs/experimental-features.md:449]: Is it worth going into any more detail here about breaking changes between 0.4 and 0.5 (in particular the DHT switch from advertising CIDs to multihashes makes 0.4 and 0.5 IPNS over PubSub peers unlikely to find each other)?

File: docs/experimental-features.md [hunk 2]
```diff
@@ -439,20 +439,22 @@ When it is enabled:
...
 
 run your daemon with the `--enable-namesys-pubsub` flag; enables pubsub.
 
 ### Road to being a real feature
 
 - [ ] Needs more people to use and report on how well it works
-- [ ] Add a mechanism for last record distribution on subscription,
-  so that we don't have to hit the DHT for the initial resolution.
-  Alternatively, we could republish the last record periodically.
+- [ ] Pubsub enabled as a real feature
```
Comment (aschmahmann) [inline docs/experimental-features.md:457]: @Stebalien Is there also a requirement here that we have a plan for dealing with people using different versions of the protocol when there are future upgrades, or is that just a standard issue we have to deal with even for real features?

File: docs/experimental-features.md [hunk 3]
```diff
@@ -427,7 +427,7 @@ ipfs config --json Experimental.ShardingEnabled true
 
 ### In Version
 
-0.4.14
+0.5.0
```
[the remark this answers is not in the document]
Comment (Stebalien) [reply docs/experimental-features.md:430]: I'd have a version history, honestly.

File: docs/experimental-features.md (hunk 1 above)
Comment (Stebalien) [reply docs/experimental-features.md:449]: Eh, not really.

File: docs/experimental-features.md (hunk 2 above)
Comment (Stebalien) [reply docs/experimental-features.md:457]: Eh, that's a standard issue.

File: docs/experimental-features.md [hunk 4]
```diff
@@ -427,7 +427,14 @@ ipfs config --json Experimental.ShardingEnabled true
...
 
-0.4.14
+0.4.14 :
+  - Introduced
+
+0.5.0 : 
+   - No longer needs to use the DHT for the first resolution
+   - When discovering PubSub peers via the DHT, the DHT key is different from previous versions
+      - This leads to 0.5 IPNS pubsub peers and 0.4 IPNS pubsub peers not being able to find each other in the DHT
+   - Robustness improvements
```
Comment (aschmahmann) [inline docs/experimental-features.md:437]: @Stebalien is this the type of version history you were looking for?

Comment (Stebalien) [reply docs/experimental-features.md:437]: lgtm

**Pull request merged**